docs/source/guides/schema-design.md

```diff
@@ -0,0 +1,198 @@
+---
+title: Schema design
+---
+
+GraphQL schemas are at their best when they are designed around the needs of client applications.  When a team is building their first GraphQL schema, they might be tempted to create literal mappings on top of existing database collections or tables using CRUD-like root fields but it’s important to consider how this could be disadvantageous.
+
+While this literal database-to-schema mapping may be a fast way to get up and running, we strongly suggest avoiding it and instead building the schema around based on how the GraphQL API will be used by the front-end.
+
+## Style conventions
+
+The GraphQL specification is flexible in the style that it dictates and doesn't impose specific naming guidelines. In order to facilitate development and continuity across GraphQL deployments, we suggest the following style conventions:
+
+* **Fields**: are recommended to be written in `camelCase`, since the majority of consumers will be client applications written in JavaScript.
+* **Types**: should be `PascalCase`.
+* **Enums**: should have their name in `PascalCase` and their values in `ALL_CAPS` to denote their special meaning.
+
+## Wrapping documents with gql
```

Comment:
This seems out of place in this article, since it's the only thing that has to do with the code. maybe we should have a separate article about schema implementation, with this and maybe schema organization